cmd/gateway/s3/gateway-s3.go

```diff
@@ -510,7 +510,7 @@ func (l *s3Objects) AbortMultipartUpload(ctx context.Context, bucket string, obj
 
 // CompleteMultipartUpload completes ongoing multipart upload and finalizes object
 func (l *s3Objects) CompleteMultipartUpload(ctx context.Context, bucket string, object string, uploadID string, uploadedParts []minio.CompletePart, opts minio.ObjectOptions) (oi minio.ObjectInfo, e error) {
-	err := l.Client.CompleteMultipartUpload(bucket, object, uploadID, minio.ToMinioClientCompleteParts(uploadedParts))
+	_, err := l.Client.CompleteMultipartUpload(bucket, object, uploadID, minio.ToMinioClientCompleteParts(uploadedParts))
```

Comment:
We can use this ETag returned and fill it in minio.ObjectInfo, rather than doing GetObjectInfo? It optimizes the call here.

Reply:
This is already being done in another PR. That was why I did not do it here. Will go ahead and add that code here.